Dodano Model

OkamiKiraa opened the pull request:
- dodano Book
- dodano Library 
- dodano View
- dodano BookView
- dodano Library View
-  Dodano Controller

Refaktoryzacja całego kodu.

OkamiKiraa committed:
Dodano Model
- dodano Book
- dodano Library
Dodano View
- dodano BookView
- Lirary View
Dodano Controller

Refaktoryzacja całego kodu.

OkamiKiraa committed:
Create README.md

OkamiKiraa committed:
Update README.md

OkamiKiraa committed:
- Refaktoryzacja ,poprawa struktury.
Klasa Main
- przejęcie kontroli nad główną pętlą programu,
- wywoływanie metod kontrolera i wyświetlanie wyników zwróconych przez kontroler/widok.

Klasa LibraryController
- usunięcie całej logiki związanej z interfejsem (brak Scanner i System.out).
- metody nie wyświetlają danych, lecz zwracają przetworzone wyniki (np. String z komunikatem, List<Book>).
- działa jako pośrednik między Main a Library.

LibraryView
- usunięcie metod drukujących na konsolę (zmiana printMenu na getMenu zwracające String).
- odpowiedzialność ograniczona wyłącznie do formatowania danych na tekst.

Inne zmiany
- Zastosowanie `stream().toList()`.

OkamiKiraa committed:
Merge remote-tracking branch 'origin/refaktor-maina' into refaktor-maina

OkamiKiraa committed:
Update README.md

blackarasu commented on README.md:
```diff
@@ -0,0 +1,27 @@
+- odpowiedzialność ograniczona wyłącznie do formatowania danych na tekst.
+
+Inne zmiany
+- Zastosowanie `stream().toList()`.
```
Rozumiem, że chcesz opisywać mi co zmieniłeś - będzie fajnie jak takie informacje będziesz zawierał w wiadomościach commita, a w readme będziesz miał:
* opis aplikacji tj. 
** co robi, jak ją uruchomić (instrukcja uruchomienia), 
** do czego służy, jakich technologii użyłeś (w twoim dotychczasowym przypadku będzie to wersja Javy, a w przyszłości jaka baza w jakiej wersji, jaki framework, specyfiki) 

przed mergem poproszę Ciebie byś przywrócił lub zastosował się do takiego wypełnienia jak opisałem powyżej

blackarasu commented on src/Main.java:
```diff
@@ -1,205 +1,105 @@
+                case "2" -> returnBook(controller, scanner);
+                case "3" -> searchBooks(controller, scanner);
+                case "4" -> addBook(controller, scanner);
+                case "5" -> {
```
Proszę przywrócić stałe opisujące jakie menu wywołujesz - nie chcemy "magicznych stringów". 

Wyobraź sobie teraz sytuację, że masz takiego switcha (chodzi o wartości oczekiwane) i nagle przychodzi Ci sytuacja, że musisz jednak zmienić te wartości. Iiii ... będziesz szukał wystąpienia w całym projekcie? Chyba nie, bo to prowadzi do sytuacji, gdzie możesz zapomnieć lub nie wiedzieć o jakimś przypadku, a to prowadzi do babola i kompromitacji dla zespołu w pewnym sensie

czyli w skrócie jedyne co trzeba zmienić to 
```
case OPTION_BORROW -> borrowBookMenu(scanner, library);
                case OPTION_RETURN -> returnBookMenu(scanner, library);
                case OPTION_SEARCH -> searchBooksMenu(scanner, library);
                case OPTION_ADD -> addBookMenu(scanner, library);
                case OPTION_EXIT ->
 ```

OkamiKiraa replied:
done

blackarasu commented on src/Main.java:
```diff
@@ -1,205 +1,105 @@
+import java.util.List;
+import java.util.Scanner;
 
+public class Main {
```
odstęp pustej lini pomiędzy otwarciem klamry klasy a pierwszym elementem (w tym przypadku metody)

OkamiKiraa replied:
done

blackarasu commented on src/Main.java:
```diff
@@ -1,205 +1,105 @@
-                    System.out.println("Nieprawidłowe dane.");
-                }
-            }
+        if (bookIndex > 0 && bookIndex <= borrowedBooks.size()) {
```
taki warunek Ci się często powtarza - przerzuć go do osobnej metody statycznej opisującej co sprawdza (np. isValidIndex)

OkamiKiraa replied:
done

blackarasu commented on src/Main.java:
```diff
@@ -1,205 +1,105 @@
-    private static void confirmAndBorrow(Scanner scanner, Library library, Book book) {
-        if (library.isBorrowed(book)) {
-            System.out.println(String.format("Ta książka jest już wypożyczona: %s", book));
+        if (bookIndex > 0 && bookIndex <= results.size()) {
```
taki warunek Ci się często powtarza - przerzuć go do osobnej metody statycznej opisującej co sprawdza (np. isValidIndex)

OkamiKiraa replied:
done

blackarasu commented on src/Main.java:
```diff
@@ -1,205 +1,105 @@
+    private static void addBook(LibraryController controller, Scanner scanner) {
         System.out.print("Podaj tytuł nowej książki: ");
-        String newTitle = scanner.nextLine();
+        String title = scanner.nextLine();
```
ok zmiana, w teorii tamta nazwa zmienny nie była zła - ludzie nazywają z prefixem new dla obiektów / pól które są przygotowane do stworzenia nowych obiektów

OkamiKiraa replied:
Pomógł mi wujek google , reddit, AI, wiedza ogólna została połączona

blackarasu commented on src/Main.java:
```diff
@@ -1,205 +1,105 @@
+        String title = scanner.nextLine();
         System.out.print("Podaj autora nowej książki: ");
-        String newAuthor = scanner.nextLine();
+        String author = scanner.nextLine();
```
ok zmiana, w teorii tamta nazwa zmienny nie była zła - ludzie nazywają z prefixem new dla obiektów / pól które są przygotowane do stworzenia nowych obiektów

OkamiKiraa replied:
Pomógł mi wujek google , reddit, AI, wiedza ogólna została połączona

blackarasu commented on src/Main.java:
```diff
@@ -1,205 +1,105 @@
     }
 
-    private static void importedBooks(Library library) {
+    private static void importBooks(Library library) {
```
fajnie byłoby już tutaj przygotować by stworzyć Interfejs który będzie definiował taką metodę i stworzyć klasę implementującą taki interfejs przy użyciu wnętrza tej metody oraz użyć taką klasę przy starcie aplikacji 

taka zmiana powinna Ci ułatwić późniejszą pracę z tym zadaniem co chciałeś zrobić

OkamiKiraa replied:
Ok - zrobię, jednakże najpierw poprawię resztę, merguje, pushnę, a potem zabiorę się za Twoje polecenie w następnym commicie.

blackarasu replied:
To będzie zrobione w innym tasku

blackarasu commented on src/controller/LibraryController.java:
```diff
@@ -0,0 +1,64 @@
+
+import java.util.List;
+
+public class LibraryController {
+    private final Library library;
```
odstęp pustej lini pomiędzy otwarciem klamry klasy a pierwszym elementem (w tym przypadku pola)

OkamiKiraa replied:
done

blackarasu commented on src/model/Book.java:
```diff
@@ -0,0 +1,33 @@
+import java.util.Objects;
+
+public class Book {
+    private final String title;
```
odstęp pustej lini pomiędzy otwarciem klamry klasy a pierwszym elementem

OkamiKiraa replied:
done

blackarasu commented on src/model/Library.java:
```diff
@@ -0,0 +1,43 @@
+import java.util.*;
+
+public class Library {
+    private final Set<Book> books = new HashSet<>();
```
odstęp pustej lini pomiędzy otwarciem klamry klasy a pierwszym elementem (w tym przypadku pola)

OkamiKiraa replied:
done

blackarasu commented on src/view/BookView.java:
```diff
@@ -0,0 +1,16 @@
+import model.Book;
+
+public class BookView {
+    public static String showBook(Book book) {
```
odstęp pustej lini pomiędzy otwarciem klamry klasy a pierwszym elementem (w tym przypadku metody)

OkamiKiraa replied:
done

blackarasu commented on src/view/LibraryView.java:
```diff
@@ -0,0 +1,44 @@
+// bookview może być wykorzystane w library view w showbooks, kolekcja jest literowalna i da się ją wyciągnąć,
+// pozbądź się tego printLN , view ma mi zwrócić stringa - albo coś , żeby użytkownik mógł łatwo przerzucić, just STRINGS.
+// zamiast przerzucić na array ,
+// mam to naprawić wszystko. na bracha refactor maina.
```
pamiętaj by usunąć taki niepotrzebny komentarz lub go nie commitować (wybiurczy commit)

OkamiKiraa replied:
done

blackarasu commented on src/view/BookView.java:
```diff
@@ -0,0 +1,16 @@
+
+// ZMIENIC NA STRING FORMAT
+// lepiej by było, żeby zwracało z pojedyńczej, a library view powinien korzystać z bookview...
+//
```
pamiętaj by usunąć taki niepotrzebny komentarz lub go nie commitować (wybiurczy commit)

OkamiKiraa replied:
done

blackarasu commented on src/view/BookView.java:
```diff
@@ -0,0 +1,16 @@
+
+public class BookView {
+    public static String showBook(Book book) {
+        return "\"" + book.getTitle() + "\" by " + book.getAuthor();
```
proszę zamień to na String.format

OkamiKiraa replied:
done

blackarasu commented on src/model/Book.java:
```diff
@@ -0,0 +1,33 @@
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) return true;
+        if (!(obj instanceof Book other)) return false;
```
wiem, żę Java pozwala na niedawanie klamer przy jednolinijkowych warunkach, ale proszę trzymaj się jednej koncepcji - np. w LibraryController jednolinijkowy warunek dałeś w klamrze - preferowane jest użycie klamer bo zmniejszy Ci ryzyko popełnienia błędu przy dodaniu jakiejś akcji w tym warunku (z jednej linii powstaną Ci conajmniej dwie linie, zapomnisz o klamrach i dla ifa i tak wykona się tylko pierwsza akcja a druga wywoła się zawsze bo jest poza ciałem warunku)

OkamiKiraa replied:
done

blackarasu commented on src/model/Library.java:
```diff
@@ -0,0 +1,43 @@
+    }
+
+    public boolean borrowBook(Book book) {
+        if (!books.contains(book)) return false;
+        if (borrowedBooks.contains(book)) return false;
```
wiem, żę Java pozwala na niedawanie klamer przy jednolinijkowych warunkach, ale proszę trzymaj się jednej koncepcji - np. w LibraryController jednolinijkowy warunek dałeś w klamrze - preferowane jest użycie klamer bo zmniejszy Ci ryzyko popełnienia błędu przy dodaniu jakiejś akcji w tym warunku (z jednej linii powstaną Ci conajmniej dwie linie, zapomnisz o klamrach i dla ifa i tak wykona się tylko pierwsza akcja a druga wywoła się zawsze bo jest poza ciałem warunku)

OkamiKiraa replied:
done

blackarasu commented on src/controller/LibraryController.java:
```diff
@@ -0,0 +1,64 @@
+}
+
+
+// senpai mówi, że coś mi nie wyszło, więc... chodzi o to, że nazywam klasy library controller , czyli chciałbym zarządzać library, a nasze library ma books i borrow books, a to co tu mam, to tak na serio te metody oznaczają
+// w library controller, ma zarządzać biblioteką, a biblioteka zarządza też książkami, bo ma swoje książki , więc my chcemy w library controller chcemy mieć operacje, aby je wykonać na bibliotece, a w mainie mieć pętle z zapytaniami do menu, bo teraz dałem , że library controller jest jednocześnie kontrolerem i viewierem (lol)
+// nie robić app controller i app view, bo to ponad miarę (hmm), ale chodzi o to, że jak mam pętle RUN to ma być w MAIN,
+// library controller powinien mieć połączenie z VIEW... hmm, klaps senpaiowi w tyłek? :)))))))))
+// nie robić konstruktorem array list , tylko toarray, a jak chce mieć to list, stream.tolist
+
+// REBASE , conflicts - ogarnąć w gicie... konflikty mogłyby się mocno gryźć, założeniem jest to, że zaczynając nowego taska nie zaciągam zmian ()w diealny mświecie), tylko zaciągam od dev branchu, nasz main/master jest naszym branchem releasowanym, żeby nie było kłopotu
+// przerzucić METODE RUN I WYJEBAĆ Z LIBRARY CONTRELER bede mial , Z PRINT MENU fajnie byłoby dać do view, (senpai mówi, ja robię), ale raczej do maina w metodzie, metody poniżej print menu, wolimy zeby controller library wytrzucil w metodzie return, a do main co mam na samym początku ze scannera i systema początkowego
+// 49 linijka , 72 linia, 92 , linia, ogarnąć library controller, zamiast konstruktora to po stworzeniu przerzucić na to list.
+// dodanie zapisu i odczytu do pliku, zapis stanu library do pliku .txt albo .csv i odczyt przy starcie, dodać klasę user, zamiast na libarry borrowed books, byłby
```
pamiętaj by usunąć taki niepotrzebny komentarz lub go nie commitować (wybiurczy commit)

OkamiKiraa replied:
done

blackarasu commented on src/controller/LibraryController.java:
```diff
@@ -0,0 +1,64 @@
+    }
+
+    public List<Book> searchBooks(String query) {
+        // Zalecenie Senpai - uycie stream().toList() zamiast konstruktora ArrayList
```
pamiętaj by usunąć taki niepotrzebny komentarz lub go nie commitować (wybiurczy commit)

OkamiKiraa replied:
done

blackarasu commented on src/view/LibraryView.java:
```diff
@@ -0,0 +1,44 @@
+        for (Book book : books) {
+            sb.append(index)
+                    .append(". ")
+                    .append(BookView.showBook(book)) // Wykorzystanie BookView : (
```
pamiętaj by usunąć taki niepotrzebny komentarz lub go nie commitować (wybiurczy commit)

OkamiKiraa replied:
done

blackarasu commented on src/view/LibraryView.java:
```diff
@@ -0,0 +1,44 @@
+            3. Szukaj książek
+            4. Dodaj książkę do biblioteki
+            5. Zakończ""";
+    }
```
to do klasy main z akcesorem prywatnym

OkamiKiraa replied:
i think done

blackarasu commented on src/model/Library.java:
```diff
@@ -0,0 +1,43 @@
+    }
+
+    public Set<Book> getBorrowedBooks() {
+        return new HashSet<>(borrowedBooks);
```
borrowedBooks już jest Setem, nie musisz go przepakowywać na nowo do Seta

OkamiKiraa replied:
Done?

blackarasu commented on src/model/Library.java:
```diff
@@ -0,0 +1,43 @@
+        Set<Book> matches = new HashSet<>();
+        for (Book b : books) {
+            if (b.getTitle().toLowerCase().contains(lower) ||
+                    b.getAuthor().toLowerCase().contains(lower)) {
```
ten warunek za dużo nie mówi na pierwszy rzut oka. Przerzuć go do statycznej prywatnej metody zwracającą boolean i jako nazwa metody coś co będzie opisywać daną akcję

OkamiKiraa replied:
done

blackarasu commented on src/model/Library.java:
```diff
@@ -0,0 +1,43 @@
+    public Set<Book> searchBooks(String query) {
+        String lower = query.toLowerCase();
+        Set<Book> matches = new HashSet<>();
+        for (Book b : books) {
```
w Javie staramy się unikać jednoliterowych zmiennych (jedynym wyjątkiem są funkcje anonimowe oraz funkcje lambda). zmień proszę na nazwę opisującą

OkamiKiraa replied:
done

blackarasu commented on src/controller/LibraryController.java:
```diff
@@ -0,0 +1,64 @@
+            return "Błąd: Tytuł i autor nie mogą być puste.";
+        }
+        if (library.addBook(title, author)) {
+            return "Pomyślnie dodano książkę: \"" + title + "\"";
```
zamień na String format

OkamiKiraa replied:
done

blackarasu commented on src/controller/LibraryController.java:
```diff
@@ -0,0 +1,64 @@
+
+    public String returnBook(Book book) {
+        if (library.returnBook(book)) {
+            return "Pomyślnie zwrócono: " + BookView.showBook(book);
```
zamień na String.format

OkamiKiraa replied:
done

blackarasu commented on src/controller/LibraryController.java:
```diff
@@ -0,0 +1,64 @@
+            return "Błąd: Ta książka jest już wypożyczona.";
+        }
+        if (library.borrowBook(book)) {
+            return "Pomyślnie wypożyczono: " + BookView.showBook(book);
```
Zamień na String.format

OkamiKiraa replied:
done

blackarasu reviewed (changes requested):
Małe głupotki

OkamiKiraa committed:
- dodano stałe dla opcji menu eliminując "magiczne stringi",

- wprowadzono metodę isValidChoice() zastępującą powtarzający się warunek,

- przeniesiono getMenu() z LibraryView do Main,

- zastąpiono konkatenację stringów przez String.format(),

- dodano klamry do jednolinijkowych instrukcji dla spójności stylu,

- zmieniono jednoliterową zmienną b na opisową book,

- wydzielono złożony warunek do metody bookMatchesQuery(),

- usunięto defensive copy w getBorrowedBooks() zwracając bezpośrednio Set,

OkamiKiraa committed:
Merge remote-tracking branch 'origin/refaktor-maina' into refaktor-maina

OkamiKiraa committed:
Merge branch 'master' into refaktor-maina

OkamiKiraa committed:
Update README.md

blackarasu commented on src/Main.java:
```diff
@@ -12,194 +16,110 @@ public class Main {
-    private static void loop(Scanner scanner, Library library) {
         while (true) {
-            printMenu();
+            System.out.println(getMenu());
```
skoro już jest to w main to może być ten println w metodzie jak wcześniej printMenu jak. jak coś to tylko notka a nie review

OkamiKiraa committed:
- dodano stałe dla opcji menu eliminując "magiczne stringi",

- wprowadzono metodę isValidChoice() zastępującą powtarzający się warunek,

- przeniesiono getMenu() z LibraryView do Main,

- zastąpiono konkatenację stringów przez String.format(),

- dodano klamry do jednolinijkowych instrukcji dla spójności stylu,

- zmieniono jednoliterową zmienną b na opisową book,

- wydzielono złożony warunek do metody bookMatchesQuery(),

- usunięto defensive copy w getBorrowedBooks() zwracając bezpośrednio Set,

OkamiKiraa committed:
Merge remote-tracking branch 'origin/refaktor-maina' into refaktor-maina

OkamiKiraa committed:
- refaktoryzacja, kodu
- zmienione nazewnictwo i dodane brakujące nielogiczne sformułowania,

blackarasu commented:
Approved and squashed and pushed also deleted remote branch refaktor-maina
